tests/e2e.py

```diff
@@ -0,0 +1,204 @@
+# -*- coding: utf-8 -*-
+# Copyright 2020 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""End to end test for the gcp module."""
+
+from googleapiclient.errors import HttpError
+
+from libcloudforensics import gcp
+
+import unittest
+import logging
+import json
+import time
+
+log = logging.getLogger()
+
+
+class EndToEndTest(unittest.TestCase):
+  """End to end test on GCP.
+  Test that gcp.py is able to correctly make a copy of the boot disk and of another given disk
+  attached to a given instance. The copies are then attached to another instance to run
+  forensics on the disk copies.
+  Add your project information to a project.info file:
+  {
+    "project_id": "xxx",
+    "instance": "xxx",
+    "disk": "xxx", # optional
+    "zone": "xxx"
+  }
+  """
+
+  def setUp(self):
+    super(EndToEndTest, self).setUp()
+    self.boot_disk_copy = None
+    self.disk_to_forensic_copy = None
+    self.analysis_vm = None
+    self.analysis_vm_name = 'new-vm-for-analysis'
+    try:
+      with open('project.info') as json_file:
+        data = json.load(json_file)
+        self.project_id = data['project_id']
```

Comment:
You can probably "close" the file here after you've opened and loaded it. This would allow you to move the exception management up and group code by functionality